### ISSUE TITLE
Example for monotonic counters?

### ISSUE THREAD

**Issue by rcarmo** [MicrosoftDocs/LogAnalyticsExamples#18]
I'm trying to plot "Network Total Bytes" (which is a monotonic counter), and need a way to compute the deltas inside time bins and extrapolate total traffic per bin (or per second).

Any simple way to do that?

**Comment by noakup** [comment]
Hi,
not sure what exactly you mean, perhaps that will help:
`WireData
| summarize avg(TotalBytes) by bin(TimeGenerated, 1h) 
| render timechart `

You're welcome to raise more questions on our [community space](https://techcommunity.microsoft.com/t5/Azure-Log-Analytics/bd-p/AzureLogAnalytics) where similar Qs have been raised and answered.

Thanks

**Comment by rcarmo** [comment]
Nope, doesn't help. What I'm saying is that I want to chart things that increase continuously (like packet counters) on a per-interval basis (i.e., plot only differences between successive values over time, and not the absolute value).

**Comment by rs38** [comment]
@rcarmo isn't there always a counter based a certain timespan like total xy per second?!

**Comment by rcarmo** [comment]
@rs38 No, per-interval counters aren't always available. And when dealing with networking, it is often critical to be able to be able to bin the data for cross-referencing with other event types, and avoid averages (which is why I opened this ticket in the first place).

To this date, I have not yet been able to do this solely inside Log Analytics - I've had to export the data and post-process it.

**Comment by rs38** [comment]
@rcarmo this should work for you. I'm curious if there are some more elegant ways….?!
```
let t1 = Perf 
| where CounterName == "Requests Failed Total"
| project Tg1 = TimeGenerated , C1 = CounterValue 
| sort by Tg1 desc 
| extend rn = row_number(); 
let t2 = Perf
| where CounterName == "Requests Failed Total"
| project Tg2 = TimeGenerated , C2 = CounterValue 
| sort by Tg2 desc 
| extend rn = row_number() + 1;
t1
| join (t2) on rn
| project  Tg1, C2-C1
```

**Comment by jruales** [comment]
@rcarmo if the time bins are regular, you can try a [make-series](https://docs.loganalytics.io/docs/Language-Reference/Tabular-operators/make-series-operator) to turn the data points into an array. Then, you can apply a finite impulse response filter with [series-fir](https://docs.loganalytics.io/docs/Language-Reference/Scalar-functions/series_fir()) to compute the differences between immediate rows. For this, your finite impulse response filter should be something like dynamic([-1, 1]) or dynamic([1, -1]). Finally, if you would like to turn the resulting array into a table, you can use [mvexpand](https://docs.loganalytics.io/docs/Language-Reference/Tabular-operators/mvexpand-operator). If you do end up using mvexpand, make sure that in the in the mvexpand statement you set the "limit" clause to a high number, since by default it truncates at 128 rows.

That was one option. Another option is to try an approach similar to what @rs38 recommended, or you can experiment with the [prev](https://docs.loganalytics.io/docs/Language-Reference/Window-functions/prev()) or [next](https://docs.loganalytics.io/docs/Language-Reference/Window-functions/next()) functions.

**Comment by rs38** [comment]
@jruales Wow!  I guess only knowing about the exitance of the function prev() did the trick! it would be very helpful to have something that helps finding these by translating SQL command. in this case LAG....
